### ISSUE TITLE
Feature Request - Query number of subscribers for subject

### ISSUE THREAD

**Issue by ronag** [nats-io/gnatsd#444]
I have a situation where, “if there is a provider/subscriber for a specific subject send a request otherwise read from database".

The only way to do this currently is to, "send request, if timeout then read from database". Which his fine, however it introduces an extra unnecessary latency in the case where there is no provider/subscriber for the subject.

What I'd like is a way to query NATS whether there is a subscriber to a subject, and if not just do the database request straight away.

Basically what we're implementing is a distributed cache where the cache nodes communicate through NATS and fallback to database if no node has a specific record in memory. Each node has a subscriber to a `READ.${id}` subject for every record it has in it's local cache so that other nodes can easily requests records from each other.

https://github.com/nxtedition/deepstream.io/blob/master/src/record/record-handler.js#L147

**Comment by ronag** [comment]
Note that the query does NOT have to be accurate in my specific scenario because I would just fallback to the database read. Of course false positives would be better than false negatives but it doesn't really matter that much.

**Comment by ronag** [comment]
Trying to explain the scenario a bit better.

We have a realtime websocket layer on top of couchdb. Where each websocket server keeps an in memory LRU cache of the records that have been requested and read from the database.

Since we have several such websocket nodes which communicate with different clients their caches can contain different records. So what I want to achieve is that "if record is not in local cache, try to ask the other nodes and as a last resort ask the database".

However, as far as I can see there are only 2 ways to achieve this:

A. Broadcast a `READ.{ID}` and wait for a hopeful response from ONE of the nodes. If no response is received within a timeout fetch from the database.

B. Somehow find out if a node has the record in it's cache and avoid the timeout. However, that would require that you know how many subscribers there is on a topic so that, e.g. `READ` and when received a `NO` from all of subscribers read from the database.
